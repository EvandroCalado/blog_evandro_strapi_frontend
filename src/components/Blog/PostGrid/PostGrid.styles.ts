import styled, { css } from 'styled-components';
import { Title } from '../../Shared/Heading/Heading.styles';

export const Container = styled.div`
  ${({ theme }) => css`
    width: 100%;
    min-height: 49.1rem;
    max-width: ${theme.frameSizes['6xl']};
    margin: 0 auto;

    & > ${Title} {
      display: inline-block;
      font-size: ${theme.font.sizes.xl};
      padding: 0 ${theme.spacings.md};
    }
`}
`;

export const GridContainer = styled.div`
  ${({ theme }) => css`
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(32rem, 1fr));
    gap: ${theme.spacings.sm};
    width: 100%;
    max-width: ${theme.frameSizes['6xl']};
    margin: 0 auto;
`}
`;
