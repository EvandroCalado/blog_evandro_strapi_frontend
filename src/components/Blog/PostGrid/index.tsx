import { Posts as StrapiPosts } from '../../../types/strapi';
import PostCard from '../../Blog/PostCard';
import PostLatest from '../../Blog/PostLatest';
import Heading from '../../Shared/Heading';
import PostNotFound from '../PostNotFound';
import * as Styled from './styles';

export interface PostGridProps {
  posts: StrapiPosts;
  gridTitle?: string;
}

export default function PostGrid({
  posts,
  gridTitle = 'Postagens Recentes',
}: PostGridProps) {
  if (posts.data.length === 0) {
    return <PostNotFound />;
  }

  const postsSlice = posts.data.slice(1);

  return (
    <Styled.Container>
      <Heading>{gridTitle}</Heading>
      <Styled.GridContainer>
        {<PostLatest post={posts.data[0]} />}
        {postsSlice.map((post) => (
          <PostCard key={post.id} post={post} />
        ))}
      </Styled.GridContainer>
    </Styled.Container>
  );
}
