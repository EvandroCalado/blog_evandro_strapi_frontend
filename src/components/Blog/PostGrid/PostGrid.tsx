import { Posts as StrapiPosts } from '../../../types/strapi';
import Heading from '../../Shared/Heading/Heading';
import PostCard from '../PostCard/PostCard';
import PostLatest from '../PostLatest/PostLatest';
import PostNotFound from '../PostNotFound/PostNotFound';
import * as Styled from './PostGrid.styles';

export interface PostGridProps {
  posts: StrapiPosts;
  gridTitle?: string;
}

export default function PostGrid({
  posts,
  gridTitle = 'Postagens Recentes',
}: PostGridProps) {
  if (posts.data.length === 0) {
    return <PostNotFound />;
  }

  const postsSlice = posts.data.slice(1);

  return (
    <Styled.Container>
      <Heading>{gridTitle}</Heading>
      <Styled.GridContainer>
        {<PostLatest post={posts.data[0]} />}
        {postsSlice.map((post) => (
          <PostCard key={post.id} post={post} />
        ))}
      </Styled.GridContainer>
    </Styled.Container>
  );
}
