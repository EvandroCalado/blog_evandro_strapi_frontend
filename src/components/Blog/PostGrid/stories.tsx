import { Meta, StoryFn } from '@storybook/react';
import PostGrid, { PostGridProps } from '.';

import mock from '../../../utils/db.json';
const posts = mock.data.posts;

export default {
  title: 'Components/Blog/PostGrid',
  component: PostGrid,
  args: {
    posts,
  },
  parameters: {
    layout: 'fullscreen',
  },
} as Meta;

export const Default: StoryFn<PostGridProps> = (args) => <PostGrid {...args} />;
