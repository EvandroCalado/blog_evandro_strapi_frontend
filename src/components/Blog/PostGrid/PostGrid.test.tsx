import '@testing-library/jest-dom';
import { screen } from '@testing-library/react';
import { renderTheme } from '../../../styles/render-theme';
import PostGrid from './PostGrid';

import mock from '../../../utils/db.json';
const posts = mock.data.posts;

describe('<PostGrid />', () => {
  it('should render with default values', () => {
    renderTheme(<PostGrid posts={posts} />);
    expect(
      screen.getByRole('heading', { name: 'Postagens Recentes' }),
    ).toBeInTheDocument();
  });

  it('should render with GridTitle', () => {
    renderTheme(<PostGrid posts={posts} gridTitle="test" />);
    expect(screen.getByRole('heading', { name: 'test' })).toBeInTheDocument();
  });

  it('should render without posts', () => {
    const mockPosts = {
      data: [],
      meta: {
        pagination: {
          total: 0,
          page: 1,
          pageSize: 10,
          pageCount: 1,
        },
      },
    };

    renderTheme(<PostGrid posts={mockPosts} gridTitle="test" />);
    expect(
      screen.getByRole('heading', { name: 'Post nÃ£o encontrado !' }),
    ).toBeInTheDocument();
  });

  it('should render a snapshot', () => {
    const { container } = renderTheme(<PostGrid posts={posts} />);
    expect(container).toMatchSnapshot();
  });
});
