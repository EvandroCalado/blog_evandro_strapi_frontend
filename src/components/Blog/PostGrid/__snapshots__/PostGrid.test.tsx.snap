// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostGrid /> should render a snapshot 1`] = `
.c2 {
  text-transform: none;
}

.c9 {
  display: flex;
  align-content: center;
  gap: 1rem;
}

.c9 svg {
  width: 2rem;
}

.c10 {
  display: flex;
  align-items: end;
  font-weight: 600;
}

.c11 {
  display: flex;
  align-items: center;
  flex-direction: column;
  padding: 3rem 2rem;
}

.c11>a {
  width: 100%;
  height: 20rem;
  margin-bottom: 2rem;
}

.c12 {
  position: relative;
  width: 100%;
  height: 100%;
  margin-bottom: 2rem;
}

.c12 img {
  object-fit: cover;
  border-radius: 1rem;
}

.c13 {
  display: flex;
  flex-direction: column;
  gap: 2rem;
}

.c13>a {
  transition: all 200ms ease-in-out;
}

.c13>a:hover {
  text-decoration: underline;
  filter: invert(.2);
}

.c13 .c1 {
  overflow: hidden;
  display: -webkit-box;
  -webkit-box-orient: vertical;
  -webkit-line-clamp: 2;
}

.c14 {
  margin: 0.5rem 0;
  overflow: hidden;
  display: -webkit-box;
  -webkit-box-orient: vertical;
  -webkit-line-clamp: 4;
}

.c15 {
  display: flex;
  align-items: center;
  justify-content: space-between;
}

.c4 {
  display: flex;
  align-items: center;
  gap: 4rem;
  padding: 3rem 2rem;
  grid-column: 1/4;
}

.c4>a {
  width: 100%;
  height: 35rem;
}

.c5 {
  position: relative;
  width: 100%;
  height: 100%;
}

.c5 img {
  object-fit: cover;
  border-radius: 1rem;
}

.c6 {
  width: 100%;
  display: flex;
  flex-direction: column;
  gap: 3rem;
}

.c6>a {
  transition: all 200ms ease-in-out;
}

.c6>a:hover {
  text-decoration: underline;
  filter: invert(.2);
}

.c7 {
  margin: 3rem 0;
  overflow: hidden;
  display: -webkit-box;
  -webkit-box-orient: vertical;
  -webkit-line-clamp: 5;
}

.c8 {
  display: flex;
  align-items: center;
  justify-content: space-between;
}

.c0 {
  width: 100%;
  max-width: 120rem;
  margin: 0 auto;
}

.c0>.c1 {
  display: inline-block;
  font-size: 4rem;
  padding: 0 2rem;
}

.c3 {
  display: grid;
  grid-template-columns: repeat(auto-fill, minmax(32rem, 1fr));
  gap: 1rem;
  width: 100%;
  max-width: 120rem;
  margin: 0 auto;
}

@media screen and (max-width: 992px) {
  .c4 {
    grid-column: 1/3;
  }
}

@media screen and (max-width: 768px) {
  .c4 {
    flex-direction: column;
    grid-column: 1/3;
  }
}

@media screen and (max-width: 576px) {
  .c4 {
    flex-direction: column;
    grid-column: 1/2;
  }
}

@media screen and (max-width: 768px) {
  .c5 {
    height: 30rem;
  }
}

@media screen and (max-width: 768px) {
  .c7 {
    margin: 1rem 0;
    -webkit-line-clamp: 4;
  }
}

@media screen and (max-width: 576px) {
  .c7 {
    margin: 1rem 0;
    -webkit-line-clamp: 3;
  }
}

<div>
  <div
    class="c0"
  >
    <h1
      class="c1 c2"
    >
      Postagens Recentes
    </h1>
    <div
      class="c3"
    >
      <div
        class="c4"
      >
        <a
          href="/post/desenvolvimento-de-ap-is-e-cms-flexiveis-e-eficient"
        >
          <div
            class="c5"
          >
            <img
              alt="small_pexels-lina-kivaka-14557287.jpg"
              data-nimg="fill"
              decoding="async"
              fetchpriority="high"
              sizes="(max-width: 768px) 100vw, (max-width: 1200px) 50vw, 33vw"
              src="/_next/image?url=https%3A%2F%2Fres.cloudinary.com%2Fdnhsdenji%2Fimage%2Fupload%2Fv1694225572%2Fsmall_pexels_lina_kivaka_14557287_039aa5b29b.jpg&w=3840&q=75"
              srcset="/_next/image?url=https%3A%2F%2Fres.cloudinary.com%2Fdnhsdenji%2Fimage%2Fupload%2Fv1694225572%2Fsmall_pexels_lina_kivaka_14557287_039aa5b29b.jpg&w=256&q=75 256w, /_next/image?url=https%3A%2F%2Fres.cloudinary.com%2Fdnhsdenji%2Fimage%2Fupload%2Fv1694225572%2Fsmall_pexels_lina_kivaka_14557287_039aa5b29b.jpg&w=384&q=75 384w, /_next/image?url=https%3A%2F%2Fres.cloudinary.com%2Fdnhsdenji%2Fimage%2Fupload%2Fv1694225572%2Fsmall_pexels_lina_kivaka_14557287_039aa5b29b.jpg&w=640&q=75 640w, /_next/image?url=https%3A%2F%2Fres.cloudinary.com%2Fdnhsdenji%2Fimage%2Fupload%2Fv1694225572%2Fsmall_pexels_lina_kivaka_14557287_039aa5b29b.jpg&w=750&q=75 750w, /_next/image?url=https%3A%2F%2Fres.cloudinary.com%2Fdnhsdenji%2Fimage%2Fupload%2Fv1694225572%2Fsmall_pexels_lina_kivaka_14557287_039aa5b29b.jpg&w=828&q=75 828w, /_next/image?url=https%3A%2F%2Fres.cloudinary.com%2Fdnhsdenji%2Fimage%2Fupload%2Fv1694225572%2Fsmall_pexels_lina_kivaka_14557287_039aa5b29b.jpg&w=1080&q=75 1080w, /_next/image?url=https%3A%2F%2Fres.cloudinary.com%2Fdnhsdenji%2Fimage%2Fupload%2Fv1694225572%2Fsmall_pexels_lina_kivaka_14557287_039aa5b29b.jpg&w=1200&q=75 1200w, /_next/image?url=https%3A%2F%2Fres.cloudinary.com%2Fdnhsdenji%2Fimage%2Fupload%2Fv1694225572%2Fsmall_pexels_lina_kivaka_14557287_039aa5b29b.jpg&w=1920&q=75 1920w, /_next/image?url=https%3A%2F%2Fres.cloudinary.com%2Fdnhsdenji%2Fimage%2Fupload%2Fv1694225572%2Fsmall_pexels_lina_kivaka_14557287_039aa5b29b.jpg&w=2048&q=75 2048w, /_next/image?url=https%3A%2F%2Fres.cloudinary.com%2Fdnhsdenji%2Fimage%2Fupload%2Fv1694225572%2Fsmall_pexels_lina_kivaka_14557287_039aa5b29b.jpg&w=3840&q=75 3840w"
              style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
            />
          </div>
        </a>
        <div
          class="c6"
        >
          <a
            href="/post/desenvolvimento-de-ap-is-e-cms-flexiveis-e-eficient"
          >
            <h2
              class="c1 c2"
            >
              Desenvolvimento de APIs e CMS Flexíveis e Eficient
            </h2>
          </a>
          <p
            class="c7"
          >
            Strapi é um sistema de gerenciamento de conteúdo (CMS) e framework de desenvolvimento de APIs de código aberto. Ele oferece uma interface amigável para criar e gerenciar conteúdo, além de permitir que os desenvolvedores criem APIs personalizadas para suas aplicações.
          </p>
          <div
            class="c8"
          >
            <div
              class="c9"
            >
              <span
                class="c10"
              >
                <svg
                  class="icon icon-tabler icon-tabler-clock-filled"
                  fill="none"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="{1.5}"
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M0 0h24v24H0z"
                    fill="none"
                    stroke="none"
                  />
                  <path
                    d="M17 3.34a10 10 0 1 1 -14.995 8.984l-.005 -.324l.005 -.324a10 10 0 0 1 14.995 -8.336zm-5 2.66a1 1 0 0 0 -.993 .883l-.007 .117v5l.009 .131a1 1 0 0 0 .197 .477l.087 .1l3 3l.094 .082a1 1 0 0 0 1.226 0l.094 -.083l.083 -.094a1 1 0 0 0 0 -1.226l-.083 -.094l-2.707 -2.708v-4.585l-.007 -.117a1 1 0 0 0 -.993 -.883z"
                    fill="currentColor"
                    stroke-width="{0}"
                  />
                </svg>
              </span>
              <span
                class="c10"
              >
                3 minutos de leitura
              </span>
            </div>
            <span
              class=""
            >
              8 de set. de 2023
            </span>
          </div>
        </div>
      </div>
      <div
        class="c11"
      >
        <a
          href="/post/desvendando-a-magia-do-desenvolvimento-web-moderno"
        >
          <div
            class="c12"
          >
            <img
              alt="small_pexels-utpal-chatterjee-16401141.jpg"
              data-nimg="fill"
              decoding="async"
              loading="lazy"
              sizes="320px"
              src="/_next/image?url=https%3A%2F%2Fres.cloudinary.com%2Fdnhsdenji%2Fimage%2Fupload%2Fv1694225572%2Fsmall_pexels_utpal_chatterjee_16401141_5048d3dc95.jpg&w=3840&q=75"
              srcset="/_next/image?url=https%3A%2F%2Fres.cloudinary.com%2Fdnhsdenji%2Fimage%2Fupload%2Fv1694225572%2Fsmall_pexels_utpal_chatterjee_16401141_5048d3dc95.jpg&w=16&q=75 16w, /_next/image?url=https%3A%2F%2Fres.cloudinary.com%2Fdnhsdenji%2Fimage%2Fupload%2Fv1694225572%2Fsmall_pexels_utpal_chatterjee_16401141_5048d3dc95.jpg&w=32&q=75 32w, /_next/image?url=https%3A%2F%2Fres.cloudinary.com%2Fdnhsdenji%2Fimage%2Fupload%2Fv1694225572%2Fsmall_pexels_utpal_chatterjee_16401141_5048d3dc95.jpg&w=48&q=75 48w, /_next/image?url=https%3A%2F%2Fres.cloudinary.com%2Fdnhsdenji%2Fimage%2Fupload%2Fv1694225572%2Fsmall_pexels_utpal_chatterjee_16401141_5048d3dc95.jpg&w=64&q=75 64w, /_next/image?url=https%3A%2F%2Fres.cloudinary.com%2Fdnhsdenji%2Fimage%2Fupload%2Fv1694225572%2Fsmall_pexels_utpal_chatterjee_16401141_5048d3dc95.jpg&w=96&q=75 96w, /_next/image?url=https%3A%2F%2Fres.cloudinary.com%2Fdnhsdenji%2Fimage%2Fupload%2Fv1694225572%2Fsmall_pexels_utpal_chatterjee_16401141_5048d3dc95.jpg&w=128&q=75 128w, /_next/image?url=https%3A%2F%2Fres.cloudinary.com%2Fdnhsdenji%2Fimage%2Fupload%2Fv1694225572%2Fsmall_pexels_utpal_chatterjee_16401141_5048d3dc95.jpg&w=256&q=75 256w, /_next/image?url=https%3A%2F%2Fres.cloudinary.com%2Fdnhsdenji%2Fimage%2Fupload%2Fv1694225572%2Fsmall_pexels_utpal_chatterjee_16401141_5048d3dc95.jpg&w=384&q=75 384w, /_next/image?url=https%3A%2F%2Fres.cloudinary.com%2Fdnhsdenji%2Fimage%2Fupload%2Fv1694225572%2Fsmall_pexels_utpal_chatterjee_16401141_5048d3dc95.jpg&w=640&q=75 640w, /_next/image?url=https%3A%2F%2Fres.cloudinary.com%2Fdnhsdenji%2Fimage%2Fupload%2Fv1694225572%2Fsmall_pexels_utpal_chatterjee_16401141_5048d3dc95.jpg&w=750&q=75 750w, /_next/image?url=https%3A%2F%2Fres.cloudinary.com%2Fdnhsdenji%2Fimage%2Fupload%2Fv1694225572%2Fsmall_pexels_utpal_chatterjee_16401141_5048d3dc95.jpg&w=828&q=75 828w, /_next/image?url=https%3A%2F%2Fres.cloudinary.com%2Fdnhsdenji%2Fimage%2Fupload%2Fv1694225572%2Fsmall_pexels_utpal_chatterjee_16401141_5048d3dc95.jpg&w=1080&q=75 1080w, /_next/image?url=https%3A%2F%2Fres.cloudinary.com%2Fdnhsdenji%2Fimage%2Fupload%2Fv1694225572%2Fsmall_pexels_utpal_chatterjee_16401141_5048d3dc95.jpg&w=1200&q=75 1200w, /_next/image?url=https%3A%2F%2Fres.cloudinary.com%2Fdnhsdenji%2Fimage%2Fupload%2Fv1694225572%2Fsmall_pexels_utpal_chatterjee_16401141_5048d3dc95.jpg&w=1920&q=75 1920w, /_next/image?url=https%3A%2F%2Fres.cloudinary.com%2Fdnhsdenji%2Fimage%2Fupload%2Fv1694225572%2Fsmall_pexels_utpal_chatterjee_16401141_5048d3dc95.jpg&w=2048&q=75 2048w, /_next/image?url=https%3A%2F%2Fres.cloudinary.com%2Fdnhsdenji%2Fimage%2Fupload%2Fv1694225572%2Fsmall_pexels_utpal_chatterjee_16401141_5048d3dc95.jpg&w=3840&q=75 3840w"
              style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
            />
          </div>
        </a>
        <div
          class="c13"
        >
          <a
            href="/post/desvendando-a-magia-do-desenvolvimento-web-moderno"
          >
            <h2
              class="c1 c2"
            >
              Desvendando a Magia do Desenvolvimento Web Moderno
            </h2>
          </a>
          <p
            class="c14"
          >
            O Next.js é uma estrutura de desenvolvimento web de código aberto construída sobre o React. Ele oferece uma variedade de recursos avançados que facilitam a criação de aplicativos web modernos e eficientes. Alguns dos principais recursos incluem:
          </p>
          <div
            class="c15"
          >
            <div
              class="c9"
            >
              <span
                class="c10"
              >
                <svg
                  class="icon icon-tabler icon-tabler-clock-filled"
                  fill="none"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="{1.5}"
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M0 0h24v24H0z"
                    fill="none"
                    stroke="none"
                  />
                  <path
                    d="M17 3.34a10 10 0 1 1 -14.995 8.984l-.005 -.324l.005 -.324a10 10 0 0 1 14.995 -8.336zm-5 2.66a1 1 0 0 0 -.993 .883l-.007 .117v5l.009 .131a1 1 0 0 0 .197 .477l.087 .1l3 3l.094 .082a1 1 0 0 0 1.226 0l.094 -.083l.083 -.094a1 1 0 0 0 0 -1.226l-.083 -.094l-2.707 -2.708v-4.585l-.007 -.117a1 1 0 0 0 -.993 -.883z"
                    fill="currentColor"
                    stroke-width="{0}"
                  />
                </svg>
              </span>
              <span
                class="c10"
              >
                4 minutos de leitura
              </span>
            </div>
            <span
              class=""
            >
              8 de set. de 2023
            </span>
          </div>
        </div>
      </div>
      <div
        class="c11"
      >
        <a
          href="/post/construindo-interfaces-modernas-e-reativas"
        >
          <div
            class="c12"
          >
            <img
              alt="small_pexels-nandu-vasudevan-10920674.jpg"
              data-nimg="fill"
              decoding="async"
              loading="lazy"
              sizes="320px"
              src="/_next/image?url=https%3A%2F%2Fres.cloudinary.com%2Fdnhsdenji%2Fimage%2Fupload%2Fv1694225572%2Fsmall_pexels_nandu_vasudevan_10920674_ceb67d8e65.jpg&w=3840&q=75"
              srcset="/_next/image?url=https%3A%2F%2Fres.cloudinary.com%2Fdnhsdenji%2Fimage%2Fupload%2Fv1694225572%2Fsmall_pexels_nandu_vasudevan_10920674_ceb67d8e65.jpg&w=16&q=75 16w, /_next/image?url=https%3A%2F%2Fres.cloudinary.com%2Fdnhsdenji%2Fimage%2Fupload%2Fv1694225572%2Fsmall_pexels_nandu_vasudevan_10920674_ceb67d8e65.jpg&w=32&q=75 32w, /_next/image?url=https%3A%2F%2Fres.cloudinary.com%2Fdnhsdenji%2Fimage%2Fupload%2Fv1694225572%2Fsmall_pexels_nandu_vasudevan_10920674_ceb67d8e65.jpg&w=48&q=75 48w, /_next/image?url=https%3A%2F%2Fres.cloudinary.com%2Fdnhsdenji%2Fimage%2Fupload%2Fv1694225572%2Fsmall_pexels_nandu_vasudevan_10920674_ceb67d8e65.jpg&w=64&q=75 64w, /_next/image?url=https%3A%2F%2Fres.cloudinary.com%2Fdnhsdenji%2Fimage%2Fupload%2Fv1694225572%2Fsmall_pexels_nandu_vasudevan_10920674_ceb67d8e65.jpg&w=96&q=75 96w, /_next/image?url=https%3A%2F%2Fres.cloudinary.com%2Fdnhsdenji%2Fimage%2Fupload%2Fv1694225572%2Fsmall_pexels_nandu_vasudevan_10920674_ceb67d8e65.jpg&w=128&q=75 128w, /_next/image?url=https%3A%2F%2Fres.cloudinary.com%2Fdnhsdenji%2Fimage%2Fupload%2Fv1694225572%2Fsmall_pexels_nandu_vasudevan_10920674_ceb67d8e65.jpg&w=256&q=75 256w, /_next/image?url=https%3A%2F%2Fres.cloudinary.com%2Fdnhsdenji%2Fimage%2Fupload%2Fv1694225572%2Fsmall_pexels_nandu_vasudevan_10920674_ceb67d8e65.jpg&w=384&q=75 384w, /_next/image?url=https%3A%2F%2Fres.cloudinary.com%2Fdnhsdenji%2Fimage%2Fupload%2Fv1694225572%2Fsmall_pexels_nandu_vasudevan_10920674_ceb67d8e65.jpg&w=640&q=75 640w, /_next/image?url=https%3A%2F%2Fres.cloudinary.com%2Fdnhsdenji%2Fimage%2Fupload%2Fv1694225572%2Fsmall_pexels_nandu_vasudevan_10920674_ceb67d8e65.jpg&w=750&q=75 750w, /_next/image?url=https%3A%2F%2Fres.cloudinary.com%2Fdnhsdenji%2Fimage%2Fupload%2Fv1694225572%2Fsmall_pexels_nandu_vasudevan_10920674_ceb67d8e65.jpg&w=828&q=75 828w, /_next/image?url=https%3A%2F%2Fres.cloudinary.com%2Fdnhsdenji%2Fimage%2Fupload%2Fv1694225572%2Fsmall_pexels_nandu_vasudevan_10920674_ceb67d8e65.jpg&w=1080&q=75 1080w, /_next/image?url=https%3A%2F%2Fres.cloudinary.com%2Fdnhsdenji%2Fimage%2Fupload%2Fv1694225572%2Fsmall_pexels_nandu_vasudevan_10920674_ceb67d8e65.jpg&w=1200&q=75 1200w, /_next/image?url=https%3A%2F%2Fres.cloudinary.com%2Fdnhsdenji%2Fimage%2Fupload%2Fv1694225572%2Fsmall_pexels_nandu_vasudevan_10920674_ceb67d8e65.jpg&w=1920&q=75 1920w, /_next/image?url=https%3A%2F%2Fres.cloudinary.com%2Fdnhsdenji%2Fimage%2Fupload%2Fv1694225572%2Fsmall_pexels_nandu_vasudevan_10920674_ceb67d8e65.jpg&w=2048&q=75 2048w, /_next/image?url=https%3A%2F%2Fres.cloudinary.com%2Fdnhsdenji%2Fimage%2Fupload%2Fv1694225572%2Fsmall_pexels_nandu_vasudevan_10920674_ceb67d8e65.jpg&w=3840&q=75 3840w"
              style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
            />
          </div>
        </a>
        <div
          class="c13"
        >
          <a
            href="/post/construindo-interfaces-modernas-e-reativas"
          >
            <h2
              class="c1 c2"
            >
              Construindo Interfaces Modernas e Reativas
            </h2>
          </a>
          <p
            class="c14"
          >
            O React é uma biblioteca JavaScript de código aberto mantida pelo Facebook e por uma comunidade ativa de desenvolvedores. Ele foi projetado para criar interfaces de usuário (UI) reutilizáveis e responsivas em aplicações web. A principal característica do React é a sua abordagem centrada em componentes, que permite dividir a UI em partes menores e reutilizáveis chamadas de "componentes".
          </p>
          <div
            class="c15"
          >
            <div
              class="c9"
            >
              <span
                class="c10"
              >
                <svg
                  class="icon icon-tabler icon-tabler-clock-filled"
                  fill="none"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="{1.5}"
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M0 0h24v24H0z"
                    fill="none"
                    stroke="none"
                  />
                  <path
                    d="M17 3.34a10 10 0 1 1 -14.995 8.984l-.005 -.324l.005 -.324a10 10 0 0 1 14.995 -8.336zm-5 2.66a1 1 0 0 0 -.993 .883l-.007 .117v5l.009 .131a1 1 0 0 0 .197 .477l.087 .1l3 3l.094 .082a1 1 0 0 0 1.226 0l.094 -.083l.083 -.094a1 1 0 0 0 0 -1.226l-.083 -.094l-2.707 -2.708v-4.585l-.007 -.117a1 1 0 0 0 -.993 -.883z"
                    fill="currentColor"
                    stroke-width="{0}"
                  />
                </svg>
              </span>
              <span
                class="c10"
              >
                5 minutos de leitura
              </span>
            </div>
            <span
              class=""
            >
              8 de set. de 2023
            </span>
          </div>
        </div>
      </div>
      <div
        class="c11"
      >
        <a
          href="/post/dominando-o-mundo-da-programacao-com-java-script"
        >
          <div
            class="c12"
          >
            <img
              alt="small_pexels-vidit-goel-12429032.jpg"
              data-nimg="fill"
              decoding="async"
              loading="lazy"
              sizes="320px"
              src="/_next/image?url=https%3A%2F%2Fres.cloudinary.com%2Fdnhsdenji%2Fimage%2Fupload%2Fv1694225572%2Fsmall_pexels_vidit_goel_12429032_16796deb17.jpg&w=3840&q=75"
              srcset="/_next/image?url=https%3A%2F%2Fres.cloudinary.com%2Fdnhsdenji%2Fimage%2Fupload%2Fv1694225572%2Fsmall_pexels_vidit_goel_12429032_16796deb17.jpg&w=16&q=75 16w, /_next/image?url=https%3A%2F%2Fres.cloudinary.com%2Fdnhsdenji%2Fimage%2Fupload%2Fv1694225572%2Fsmall_pexels_vidit_goel_12429032_16796deb17.jpg&w=32&q=75 32w, /_next/image?url=https%3A%2F%2Fres.cloudinary.com%2Fdnhsdenji%2Fimage%2Fupload%2Fv1694225572%2Fsmall_pexels_vidit_goel_12429032_16796deb17.jpg&w=48&q=75 48w, /_next/image?url=https%3A%2F%2Fres.cloudinary.com%2Fdnhsdenji%2Fimage%2Fupload%2Fv1694225572%2Fsmall_pexels_vidit_goel_12429032_16796deb17.jpg&w=64&q=75 64w, /_next/image?url=https%3A%2F%2Fres.cloudinary.com%2Fdnhsdenji%2Fimage%2Fupload%2Fv1694225572%2Fsmall_pexels_vidit_goel_12429032_16796deb17.jpg&w=96&q=75 96w, /_next/image?url=https%3A%2F%2Fres.cloudinary.com%2Fdnhsdenji%2Fimage%2Fupload%2Fv1694225572%2Fsmall_pexels_vidit_goel_12429032_16796deb17.jpg&w=128&q=75 128w, /_next/image?url=https%3A%2F%2Fres.cloudinary.com%2Fdnhsdenji%2Fimage%2Fupload%2Fv1694225572%2Fsmall_pexels_vidit_goel_12429032_16796deb17.jpg&w=256&q=75 256w, /_next/image?url=https%3A%2F%2Fres.cloudinary.com%2Fdnhsdenji%2Fimage%2Fupload%2Fv1694225572%2Fsmall_pexels_vidit_goel_12429032_16796deb17.jpg&w=384&q=75 384w, /_next/image?url=https%3A%2F%2Fres.cloudinary.com%2Fdnhsdenji%2Fimage%2Fupload%2Fv1694225572%2Fsmall_pexels_vidit_goel_12429032_16796deb17.jpg&w=640&q=75 640w, /_next/image?url=https%3A%2F%2Fres.cloudinary.com%2Fdnhsdenji%2Fimage%2Fupload%2Fv1694225572%2Fsmall_pexels_vidit_goel_12429032_16796deb17.jpg&w=750&q=75 750w, /_next/image?url=https%3A%2F%2Fres.cloudinary.com%2Fdnhsdenji%2Fimage%2Fupload%2Fv1694225572%2Fsmall_pexels_vidit_goel_12429032_16796deb17.jpg&w=828&q=75 828w, /_next/image?url=https%3A%2F%2Fres.cloudinary.com%2Fdnhsdenji%2Fimage%2Fupload%2Fv1694225572%2Fsmall_pexels_vidit_goel_12429032_16796deb17.jpg&w=1080&q=75 1080w, /_next/image?url=https%3A%2F%2Fres.cloudinary.com%2Fdnhsdenji%2Fimage%2Fupload%2Fv1694225572%2Fsmall_pexels_vidit_goel_12429032_16796deb17.jpg&w=1200&q=75 1200w, /_next/image?url=https%3A%2F%2Fres.cloudinary.com%2Fdnhsdenji%2Fimage%2Fupload%2Fv1694225572%2Fsmall_pexels_vidit_goel_12429032_16796deb17.jpg&w=1920&q=75 1920w, /_next/image?url=https%3A%2F%2Fres.cloudinary.com%2Fdnhsdenji%2Fimage%2Fupload%2Fv1694225572%2Fsmall_pexels_vidit_goel_12429032_16796deb17.jpg&w=2048&q=75 2048w, /_next/image?url=https%3A%2F%2Fres.cloudinary.com%2Fdnhsdenji%2Fimage%2Fupload%2Fv1694225572%2Fsmall_pexels_vidit_goel_12429032_16796deb17.jpg&w=3840&q=75 3840w"
              style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
            />
          </div>
        </a>
        <div
          class="c13"
        >
          <a
            href="/post/dominando-o-mundo-da-programacao-com-java-script"
          >
            <h2
              class="c1 c2"
            >
              Dominando o Mundo da Programação com JavaScript
            </h2>
          </a>
          <p
            class="c14"
          >
            JavaScript, frequentemente abreviado como JS, é uma linguagem de programação de alto nível que permite adicionar interatividade e funcionalidade a páginas web. É executado no navegador do cliente (front-end) e pode interagir com o HTML e o CSS para criar experiências ricas para o usuário.
          </p>
          <div
            class="c15"
          >
            <div
              class="c9"
            >
              <span
                class="c10"
              >
                <svg
                  class="icon icon-tabler icon-tabler-clock-filled"
                  fill="none"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="{1.5}"
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M0 0h24v24H0z"
                    fill="none"
                    stroke="none"
                  />
                  <path
                    d="M17 3.34a10 10 0 1 1 -14.995 8.984l-.005 -.324l.005 -.324a10 10 0 0 1 14.995 -8.336zm-5 2.66a1 1 0 0 0 -.993 .883l-.007 .117v5l.009 .131a1 1 0 0 0 .197 .477l.087 .1l3 3l.094 .082a1 1 0 0 0 1.226 0l.094 -.083l.083 -.094a1 1 0 0 0 0 -1.226l-.083 -.094l-2.707 -2.708v-4.585l-.007 -.117a1 1 0 0 0 -.993 -.883z"
                    fill="currentColor"
                    stroke-width="{0}"
                  />
                </svg>
              </span>
              <span
                class="c10"
              >
                4 minutos de leitura
              </span>
            </div>
            <span
              class=""
            >
              8 de set. de 2023
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
